unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1550,8 +1540,8 @@ handleDependencies hq = do
   let types = nubOrdOn snd . Name.sortByText (HQ.toText . fst) $ (join $ fst <$> results)
   let terms = nubOrdOn snd . Name.sortByText (HQ.toText . fst) $ (join $ snd <$> results)
   Cli.setNumberedArgs $
-    map (Text.unpack . Reference.toText . snd) types
-      <> map (Text.unpack . Reference.toText . Referent.toReference . snd) terms
+    map (SA.Type . snd) types
+      <> map (SA.Term . Referent.toReference . snd) terms
```

Comment:
Aside: I know this comment is "Outdated", but I was still wanting to answer it. I looked in the diff for these lines (e.g. old line 1553), but it doesn't match what's shown here, and I don't understand why it wouldn't? I see it on old line 1302.

I think the answer to your question is that until https://github.com/unisonweb/unison/pull/4907, the numbered args weren't shown to the user at all unless they ran `debug.numberedArgs`, making the question moot(?)

Reply:
> Aside: I looked in the diff for these lines (e.g. old line 1553), but it doesn't match what's shown here, and I don't understand why it wouldn't? I see it on old line 1302. I see Github's marked the snippet as "Outdated", but I don't think that would matter.

Ah, I think it’s because I squashed another commit into this one when it was in draft, but after I wrote this comment.

I had originally had `Ref`, `Term`, and `Type` as three separate `StructuredArgument` cases, but that just led to three copies of identical logic, so I removed the `Term` and `Type` cases until we can reasonably preserve that information.
 
> I think the answer to your question is that until #4907, the numbered args weren't shown to the user at all unless they ran `debug.numberedArgs`, making the question moot(?)

Ah, right. I conflated a few things here and got confused.

But the question wasn’t really moot – if `Reference`s are put into numbered args state, a command expecting a `Name` couldn’t then use a numbered arg. Whereas, if `HashQualified Name`s were put into the state, the same command could then use it.

I don’t think it’s a change for _this_ PR, but it seems like using the HQ instead of the Reference would help fix some of the existing numbered args problems.

Also, I think the pretty-printed numbered results lists (e.g., the output of `find`) could eventually lean more on the numbered args handling, rather that being constructed completely independently, which is another reason using HQ instead of a Reference would be a good change.

But I don’t know what the counter-arguments might be.

Reply:
Yes, makes sense.